concrete/src/Attribute/Key/SearchIndexer/StandardSearchIndexer.php

```diff
@@ -32,6 +32,71 @@ protected function getIndexEntryColumn(Key $key, $subKey = false)
     }
 
     /**
+     * For certain fields (eg TEXT) Doctrine uses the length of the longest column to determine what field type to use.
+     * For search indexing even if we may not currently have something long in a column,
+     * we need the longest possible column so that we don't truncate any data.
+     *
+     * @param array $options
+     *
+     * @return array
+     */
+    private function setTypeLength($options)
+    {
+        // If we have explicitly set a length, use it
+        if ($options['length']) {
+            return $options;
+        }
+        if ($options['type']->getName() == 'text') {
+            $options['length'] = 4294967295; // This forces Doctrine to use `LONGTEXT` instead of `TINYTEXT`
```

Comment:
PHP_INT_MAX depends on the # of bits the PHP has been compiled with.
Here's the value for 3v4l.org  (that clearly uses 64 bits):
https://3v4l.org/qUPOK